### PR TITLE
Fix create mutations for custom PK

### PR DESCRIPTION
**Pull request by mikepschneider**
*Description of changes:*

Updated creation mutation to work. with custom primary key.

*How did you test these changes?*
Added. unit tests

*Documentation update required?*
- [x ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java [hunk 1]
```diff
@@ -422,14 +424,35 @@ private static Map<String, Object> extractFieldLevelData(
...
         }
         return result;
     }
 
+    private static void insertForeignKeyValues(Map<String, Object> result, ModelField modelField,
+                                               Model instance, ModelSchema schema,
+                                               ModelAssociation association) throws AmplifyException {
+        final Object fieldValue = extractFieldValue(modelField.getName(), instance, schema);
+        if (modelField.isModel() && fieldValue instanceof Model) {
+            final ModelPrimaryKey<?> primaryKey = (ModelPrimaryKey<?>) ((Model) fieldValue).resolveIdentifier();
```
[the remark this answers is not in the document]
Comment (mikepschneider) [reply aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java]: Done

File: testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/BlogOwner.java [hunk 2]
```diff
@@ -38,7 +39,7 @@ public String getName() {
       return name;
   }
   
-  public Blog getBlog() {
```
Comment (poojamat) [inline testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/BlogOwner.java:41]: Would this cause tests to fail?

File: testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java [hunk 3]
```diff
@@ -13,7 +13,7 @@
  */
 
 public final class AmplifyModelProvider implements ModelProvider {
-  private static final String AMPLIFY_MODEL_VERSION = "313b773087e72b546d24c87e137d7b58";
+  private static final String AMPLIFY_MODEL_VERSION = "985164532d43fb0d6d9b1f37425538f8";
```
[the remark this answers is not in the document]
Comment (mikepschneider) [reply testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java:16]: I assume that changed when I copied over the regenerated files.    I can change it back, don't know. that it matters though.

File: testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java [hunk 4]
```diff
@@ -34,7 +34,7 @@ public static AmplifyModelProvider getInstance() {
   @Override
    public Set<Class<? extends Model>> models() {
     final Set<Class<? extends Model>> modifiableSet = new HashSet<>(
-            Arrays.<Class<? extends Model>>asList(Blog.class, Post.class, Comment.class, PostAuthorJoin.class, Author.class, BlogOwner.class)
```
[the remark this answers is not in the document]
Comment (mikepschneider) [reply testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java:37]: The Join models aren't created under the new relationship schema right?  I don't have one in the test app where I regenerated.  I've removed the file here.

File: testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/Blog.java [hunk 5]
```diff
@@ -136,23 +132,19 @@ public CopyOfBuilder copyOfBuilder() {
       createdAt);
   }
   public interface NameStep {
-    OwnerStep name(String name);
-  }
-  
-
-  public interface OwnerStep {
```
[the remark this answers is not in the document]
Comment (mikepschneider) [reply testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/Blog.java:143]: There is method `public BuildStep owner(BlogOwner owner) {`, it didn't generate any `public interface XXXStep` classes so I guess that's how they are generated now.

File: testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/BlogOwner.java (hunk 2 above)
Comment (mikepschneider) [reply testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/BlogOwner.java:41]: All oof my tests are passing

File: aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java [hunk 6]
```diff
@@ -422,14 +426,47 @@ private static Map<String, Object> extractFieldLevelData(
...
+                final ModelIdentifier<?> primaryKey = (ModelIdentifier<?>) ((Model) fieldValue).resolveIdentifier();
+                ListIterator<String> targetNames = Arrays.asList(association.getTargetNames()).listIterator();
+                Iterator<? extends Serializable> sortedKeys = primaryKey.sortedKeys().listIterator();
+
+                result.put(targetNames.next(), primaryKey.key().toString());
+
+                while (targetNames.hasNext()) {
+                    result.put(targetNames.next(), sortedKeys.next().toString());
+                }
+            } else if (((Model) fieldValue).resolveIdentifier() instanceof String) {
```
[the remark this answers is not in the document]
Comment (mikepschneider) [reply aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java]: Ok I will just toString() any object then, is. that ok?

[the remark this answers is not in the document]
Comment (poojamat) [reply aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java]: Yes, that should work.

File: testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (poojamat) [reply testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java]: Should be fine if the tests are passing.

File: testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (poojamat) [reply testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java:37]: If the tests are working then its fine.

**Review by poojamat — APPROVED** [review]

**Pull request merged**